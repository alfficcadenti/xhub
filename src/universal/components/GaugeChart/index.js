import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import ReactGaugeChart from 'react-gauge-chart';

class GaugeChart extends PureComponent {

    render() {
        const {
            title,
            value
        } = this.props;

        return (
            <ReactGaugeChart 
                id={title}
                nrOfLevels={20}
                percent={value/100} 
                style={{height: 150, width: 250}}
                colors={["#FF5F6D", "#00FF00"]}
                textColor={'#000000'}
            />
        )
    }
}

GaugeChart.defaultProps = {
    title: '',
    value: 0,
};

GaugeChart.propTypes = {
    title: PropTypes.string,
    value: PropTypes.number
};

export default GaugeChart;