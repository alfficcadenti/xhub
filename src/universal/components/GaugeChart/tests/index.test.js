import React from 'react';
import {shallow} from 'enzyme';
import chai, {expect} from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';
import GaugeChart from '../index';
import renderer from 'react-test-renderer';

chai.use(chaiJestSnapshot);

describe('<GaugeChart /> ', () => {
    beforeEach(() => {
        chaiJestSnapshot.setFilename(`${__filename}.snap`);
    });

    const props = {
        title: 'title',
        value: 96
    };

    it('renders correctly', () => {
        const wrapper = shallow(<GaugeChart {...props} />);
        expect(wrapper).to.have.lengthOf(1);
    });

    it('matches the snapshot', () => {
        chaiJestSnapshot.setTestName('matches the snapshot');
        const wrapper = renderer.create(<GaugeChart {...props} />);
        expect(wrapper).to.matchSnapshot();
    });
});