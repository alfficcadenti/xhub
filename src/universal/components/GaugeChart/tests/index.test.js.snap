// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className={undefined}
  id="title"
  style={
    Object {
      "height": 150,
      "width": 250,
    }
  }
/>
`;
